packages/mui-lab/src/Masonry/Masonry.js

```diff
@@ -80,15 +92,31 @@ export const getStyle = ({ ownerState, theme }) => {
 
   const transformer = createUnarySpacing(theme);
   const spacingStyleFromPropValue = (propValue) => {
-    const themeSpacingValue = Number(propValue);
-    const spacing = Number(getValue(transformer, themeSpacingValue).replace('px', ''));
+    let spacing;
+    let halfSpacing;
+    // in case of string/number value
+    if (
+      (typeof propValue === 'string' && !Number.isNaN(Number(propValue))) ||
+      typeof propValue === 'number'
+    ) {
+      const themeSpacingValue = Number(propValue);
+      spacing = parseToNumber(getValue(transformer, themeSpacingValue));
+      halfSpacing = spacing / 2;
+    } else {
+      spacing = propValue;
+      halfSpacing = divideStringFontSize(propValue);
+    }
+
     return {
-      margin: -(spacing / 2),
+      margin: -halfSpacing,
```

Comment:
I think let's hand over this calculation to CSS by using `calc(... / 2)`. It works with any unit.